### PR TITLE
[Feature] add model script, training recipe and trained weights of PiT

### LINKED ISSUE #419: [Models][330][transformer]pit
If this is your first time, please read our contributor guidelines: https://gitee.com/mindspore/mindspore/blob/master/CONTRIBUTING.md

**Is your feature request related to a problem? Please describe.**
https://github.com/rwightman/pytorch-image-models/blob/main/timm/models/pit.py
https://arxiv.org/abs/2103.16302

**Describe the solution you'd like**
model.py
readme
config

### PR DESCRIPTION
**Pull request by ziyi-luo**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [ ] You have added unit tests

## Motivation

The training recipe and trained weights of PiT are added.

## Test Plan

Please refer to "configs/pit/README.md" for reproducing and testing.

## Related Issues and PRs

Related Issue: close #419

### CODE REVIEW

File: configs/pit/README.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+Please download the [ImageNet-1K](https://www.image-net.org/challenges/LSVRC/2012/index.php) dataset for model training and validation.
+
+### Training
+
+* Distributed Training
+
+It is easy to reproduce the reported results with the pre-defined training recipe. For distributed training on multiple Ascend 910 devices, please run
+
+```shell
+# distrubted training on multiple GPU/Ascend devices
```
Comment (Songyuanwei) [inline configs/pit/README.md:51]: distrubted -> distributed

File: mindcv/models/pit.py [hunk 2]
```diff
@@ -0,0 +1,548 @@
...
+from .layers import Identity
+from .registry import register_model
+from .utils import load_pretrained
+
+__all__ = [
+    "pit_ti_224",
+    "pit_xs_224",
+    "pit_s_224",
+    "pit_b_224",
+    "pit_ti_distilled_224"
```
Comment (Songyuanwei) [inline mindcv/models/pit.py:26]: "pit_ti_distilled_224" -> "pit_ti_distilled_224",    句尾逗号

Comment (ziyi-luo) [reply mindcv/models/pit.py]: 已修改

File: configs/pit/README.md (hunk 1 above)
Comment (ziyi-luo) [reply configs/pit/README.md]: 已修改

File: configs/pit/pit_xs_ascend.yaml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+min_lr: 0.00001
+decay_epochs: 590
+warmup_epochs: 10
+
+# optimizer
+opt: 'adamw'
+filter_bias_and_bn: True
+momentum: 0.9
+weight_decay: 0.05
+dynamic_loss_scale: True
```
Comment (geniuspatrick) [inline configs/pit/pit_xs_ascend.yaml:57]: loss_scale_type: '

File: configs/pit/pit_xs_ascend.yaml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+drop_path_rate: 0.1
+dataset_sink_mode: True
+amp_level: 'O2'
+
+# loss
+loss: 'ce'
+label_smoothing: 0.1
+
+# lr scheduler
+scheduler: 'warmup_cosine_decay'
```
Comment (geniuspatrick) [inline configs/pit/pit_xs_ascend.yaml:46]: cosine_decay is ok

File: configs/pit/pit_xs_ascend.yaml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+# system
+mode: 0
+distribute: True
+num_parallel_workers: 8
+val_while_train: True
+
+# dataset
+dataset: 'imagenet'
+data_dir: 'path/to/imagenet'
```
Comment (geniuspatrick) [inline configs/pit/pit_xs_ascend.yaml:9]: /path/to/imagenet

File: configs/pit/pit_xs_ascend.yaml [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+mixup: 0.8
+cutmix: 1.0
+
+# model
+model: 'pit_xs_224'
+num_classes: 1000
+pretrained: False
+ckpt_path: ''
+keep_checkpoint_max: 10
+ckpt_save_dir: './ckpt/'
```
Comment (geniuspatrick) [inline configs/pit/pit_xs_ascend.yaml:35]: ./ckpt

File: mindcv/models/pit.py [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+        attn = self.attn_drop(attn)
+
+        x = self.batchmatmul(attn, v)
+        x = ops.reshape(ops.transpose(x, (0, 2, 1, 3)), (B, N, C))
+        x = self.proj(x)
+        x = self.proj_drop(x)
+        return x
+
+
+class DropPath(nn.Cell):
```
Comment (geniuspatrick) [inline mindcv/models/pit.py:149]: why not use droppath in layers

File: configs/pit/pit_xs_ascend.yaml (hunk 3 above)
Comment (ziyi-luo) [reply configs/pit/pit_xs_ascend.yaml:58]: We use the default "fixed" for loss_scale_type. And add it in yaml for clearness.

File: configs/pit/pit_xs_ascend.yaml (hunk 4 above)
Comment (ziyi-luo) [reply configs/pit/pit_xs_ascend.yaml]: It has been modified.

File: configs/pit/pit_xs_ascend.yaml (hunk 5 above)
Comment (ziyi-luo) [reply configs/pit/pit_xs_ascend.yaml]: It has been modified.

File: configs/pit/pit_xs_ascend.yaml (hunk 6 above)
Comment (ziyi-luo) [reply configs/pit/pit_xs_ascend.yaml]: It has been modified.

File: mindcv/models/pit.py (hunk 7 above)
Comment (ziyi-luo) [reply mindcv/models/pit.py:149]: we use droppath1D to specify the dim of input for the readability.

Comment (ziyi-luo) [reply mindcv/models/pit.py]: We didn't notice it before.Already changed

File: configs/pit/pit_xs_ascend.yaml (hunk 3 above)
Comment (ziyi-luo) [reply configs/pit/pit_xs_ascend.yaml]: It has been modified.

**Review by geniuspatrick — APPROVED** [review]
